pkg/controller/daemon/controller.go

```diff
@@ -406,11 +406,12 @@ func storeDaemonSetStatus(dsClient client.DaemonSetInterface, ds *extensions.Dae
 		return nil
 	}
 
+	ds.Status.DesiredNumberScheduled = desiredNumberScheduled
+	ds.Status.CurrentNumberScheduled = currentNumberScheduled
+	ds.Status.NumberMisscheduled = numberMisscheduled
+
 	var updateErr, getErr error
 	for i := 0; i <= StatusUpdateRetries; i++ {
-		ds.Status.DesiredNumberScheduled = desiredNumberScheduled
-		ds.Status.CurrentNumberScheduled = currentNumberScheduled
-		ds.Status.NumberMisscheduled = numberMisscheduled
```

Comment:
is this correct? if there was a conflict, `ds` gets refetched below, which means the status set outside the loop is lost. calling `UpdateStatus(ds)` then becomes a no-op, which doesn't seem right